### PR TITLE
import: clarify -o behavior for subdirectories

### PR DESCRIPTION
**Pull request by pmrowla**
> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

* Clarify `dvc import -o` behavior when output path contains subdirectories.
* See https://github.com/iterative/dvc/pull/4340

### CODE REVIEW

**Comment by shcheklein** [comment]
Thanks @pmrowla , really appreciate this! 

Just curious - is it the same for get?

**Comment by pmrowla** [comment]
@shcheklein For get we do a fetch + checkout and create the output subdirectory if it doesn't already exist, since that's the normal behavior for checkout.

File: content/docs/command-reference/import.md [hunk 1]
```diff
@@ -78,7 +78,8 @@ from the source repo.
 - `-o <path>`, `--out <path>` - specify a path (directory and/or file name) to
   the desired location to place the imported file in (instead of using the
   current working directory). If an existing directory is specified, the target
-  data will be placed inside.
+  data will be placed inside. If `path` contains a parent directory which does
+  not exist, this command will fail.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/import.md:82]: Sorry for late review @pmrowla, I have a Q:

> If `path` contains a parent directory

What does that mean? "Containing a parent" sounds contradictory. Parents contain you 😋

Comment (jorgeorpinel) [reply content/docs/command-reference/import.md:82]: Plus how can you contain something that does not exist? Then you don't contain it... Sorry, just confused by this.

Comment (pmrowla) [reply content/docs/command-reference/import.md:82]: If you run `dvc import -o foo/bar` and `foo/` doesn't already exist, `import` will fail

Comment (jorgeorpinel) [reply content/docs/command-reference/import.md:82]: I see, got it. I may reword a little. Thanks Peter!

**Comment by jorgeorpinel** [comment]
> For get we ... create the output subdirectory if it doesn't already exist

I'll add this note too.

BTW I get the technicality involving internal processes (checkout) but it's kind of weird that get creates them but import doesn't... And why doesn't import use checkout? Hmmmm 🤔

**Comment by pmrowla** [comment]
It's because in import we are defining an actual stage/dependency, but for get it's treated more like a standalone file

from https://github.com/iterative/dvc/pull/4340#discussion_r466158245

> We also don't makedirs in dvc run, but rather raise an error that wdir doesn't exist, so it seems it would be more consistent if we raise and error here too. Same with dvc add that shares the same logic with dvc run.

**Comment by jorgeorpinel** [comment]
Makes sense, that's relevant context for sure. My idea would be that for consistency maybe no DVC commands should makedirs... In fact probably only `get` does that. But anyway, I won't open a ticket about this, up to you to bring it up with the core team if you wish Peter 🙂

**Pull request merged**